components/vocabularies/api/src/test/java/org/phenotips/vocabulary/internal/DefaultVocabularyManagerTest.java

```diff
@@ -0,0 +1,362 @@
+
+    @Mock
+    private VocabularyTerm result5;
+
+    @Mock
+    private VocabularyTerm result6;
+
+    @Before
+    public void setUp() throws ComponentLookupException, InitializationException, NoSuchFieldException
+    {
+        MockitoAnnotations.initMocks(this);
+        this.vocabularyManager = this.mocker.getComponentUnderTest();
+
+        // Add the vocabularies to the map.
+        this.vocabularies = new HashMap<>();
+        this.vocabularies.put(HGNC_LABEL, this.hgnc);
+        this.vocabularies.put(CHEBI_LABEL, this.chebi);
+        this.vocabularies.put(HPO_LABEL, this.hpo);
+        this.vocabularies.put(ETHNICITY_LABEL, this.ethnicity);
+        this.vocabularies.put(OMIM_LABEL, this.omim);
+
+        this.logger = this.mocker.getMockedLogger();
+
+        // Mock a set of aliases for the hgnc vocabulary.
+        final Set<String> hgncAliases = new HashSet<>();
+        hgncAliases.add(HGNC_LABEL);
+        hgncAliases.add(HGNC_CAPS_LABEL);
+
+        // Mock a set of aliases for the chebi vocabulary.
+        final Set<String> chebiAliases = new HashSet<>();
+        chebiAliases.add(CHEBI_LABEL);
+        chebiAliases.add(CHEBI_CAPS_LABEL);
+
+        // Mock a set of aliases for the hpo vocabulary.
+        final Set<String> hpoAliases = new HashSet<>();
+        hpoAliases.add(HPO_LABEL);
+        hpoAliases.add(HP_LABEL);
+
+        // Mock a set of aliases for the ethnicity vocabulary.
+        final Set<String> ethnicityAliases = new HashSet<>();
+        ethnicityAliases.add(ETHNICITY_LABEL);
+        ethnicityAliases.add(ETHNO_LABEL);
+
+        // Mock a set of aliases for the omim vocabulary.
+        final Set<String> omimAliases = new HashSet<>();
+        omimAliases.add(OMIM_LABEL);
+        omimAliases.add(MIM_LABEL);
+
+        // Mock the Vocabulary#getAliases calls.
+        when(this.hgnc.getAliases()).thenReturn(hgncAliases);
+        when(this.chebi.getAliases()).thenReturn(chebiAliases);
+        when(this.hpo.getAliases()).thenReturn(hpoAliases);
+        when(this.ethnicity.getAliases()).thenReturn(ethnicityAliases);
+        when(this.omim.getAliases()).thenReturn(omimAliases);
+
+        // Mock the Vocabulary#getSupportedCategories calls.
+        when(this.hgnc.getSupportedCategories()).thenReturn(Collections.singletonList(GENE_CATEGORY));
+        when(this.chebi.getSupportedCategories()).thenReturn(Collections.singletonList(CHEMICAL_CATEGORY));
+        when(this.hpo.getSupportedCategories()).thenReturn(Arrays.asList(PHENOTYPE_CATEGORY, PHENOTYPE_Q_CATEGORY));
+        when(this.ethnicity.getSupportedCategories()).thenReturn(Collections.singletonList(ETHNICITY_CATEGORY));
+        when(this.omim.getSupportedCategories()).thenReturn(Arrays.asList(DISEASE_CATEGORY, GENE_CATEGORY));
+
+        // Set vocabularies field manually, and run initialize again to compute all other fields correctly.
+        ReflectionUtils.setFieldValue(this.vocabularyManager, VOCABULARIES_LABEL, this.vocabularies);
```

Comment:
Using `ReflectionUtils` is not recommended, and in most cases there's an easier way to achieve the goal. In this case, you can use `mocker.registerComponent` to register the mocked vocabularies, which will then be correctly injected once you call `getComponentUnderTest`.